packages/strapi-plugin-graphql/services/build-aggregation.js

```diff
@@ -252,19 +270,52 @@ const createGroupByFieldsResolver = function(model, fields) {
       ...convertToQuery(filters.where),
     };
 
-    const result = await buildQuery({
-      model,
-      filters: convertRestQueryParams(params),
-      aggregate: true,
-    }).group({
-      _id: `$${fieldKey === 'id' ? model.primaryKey : fieldKey}`,
-    });
+    if (model.orm === 'mongoose') {
+      const result = await buildQuery({
+        model,
+        filters: convertRestQueryParams(params),
+        aggregate: true,
+      }).group({
+        _id: `$${fieldKey === 'id' ? model.primaryKey : fieldKey}`,
+      });
+
+      return preProcessGroupByData({
+        result,
+        fieldKey,
+        filters,
+      });
+    }
+
+    if (model.orm === 'bookshelf') {
+      let { limit, ...aggFilters } = filters;
+      return model
+        .query(qb => {
+          // apply filters without pagination limit
+          buildQuery({ model, filters: aggFilters })(qb);
+          qb.groupBy(fieldKey);
+          qb.select(fieldKey);
+        })
+        .fetchAll()
+        .then(result => {
+          let values = result.models
+            .map(m => _.get(m, ['attributes', fieldKey])) // extract aggregate field
+            .filter(v => !!v) // remove null
+            .map(v => '' + v); // convert to string
+          return values.map(v => ({
+            key: v,
+            connection: () => {
+              return {
+                ...filters,
+                where: {
+                  ...(filters.where || {}),
+                  [fieldKey]: v,
```

Comment:
I think this will create issues if you already have a filter on the key / or add a new goupBy on the same key later on. It will not filter properly.  We will need to add feeatures in the DB layer directly at one point

Reply:
All the problems you mention will have arisen when we decide to support nested arguments in aggregate fields.

Reply:
Sorry I'm not sure I understood your answer. 

We could mitigate this by supporting an array of filters instead of an object for example :) this would accumulate the filters instead of overwritting them

Reply:
Can you provide some pseudo-code so I can understand the problem better?

Reply:
if you create a dumb query like:

```graphql
{
  restaurantsConnection( where:{ country: "value"}) {
    groupBy {
      country {
        key
        connection {
          values
        }
      }
    }
  }
}
```

The fiter for country will be overwritten at each layer as you are merging filters with :

```js
{
where: {
   ...(filters.where || {}),
  [fieldKey]: v,
}
```

if there already is fieldKey in fitlers.where it gets overwritten and merged with the previous filter

If you had an array and did 

```js
where: filters.where.concat({ [fieldKey]: v })
```

This would add on the previous filters and not break the query inside of the groupBy

In this example the values returns would filter on the country by key and not on the first `country : "value"`

Hope this makes sense :)

Reply:
I think my code work as expected, for instance

Given query
```graphql
{
  restaurantsConnection( where:{ country: "France"}) {
    groupBy {
      country {
        key
        connection {
          values {
            id
            name
            country
          }
        }
      }
    }
  }
}
```

I think the result should be:
```js
let result = {
    groupBy: {
        country: [
            {
                key: "France",
                connection: {
                    values: [
                        { id: "1", name: "some restaurant 1", country: "France" },
                        { id: "2", name: "some restaurant 2", country: "France" },
                        // { id: "3", name: "some restaurant 3", country: "Vietnam" }, // not me!
                    ]
                }
            },
        ]
    }
};
```

Reply:
Ok If this is what it returns that's great. I will be testing the PR as is ;) thanks for your contribution !